Userland/Libraries/LibDeviceTree/DeviceTree.h

```diff
@@ -18,6 +19,67 @@
 
 namespace DeviceTree {
 
+// Devicetree Specification 0.4 (DTSpec): https://github.com/devicetree-org/devicetree-specification/releases/download/v0.4/devicetree-specification-v0.4.pdf
+
+class DeviceTree;
+class Node;
+
+class Address {
+public:
+    Address() = default;
+    Address(ReadonlyBytes data)
+        : m_raw(static_cast<decltype(m_raw)>(data))
+    {
+    }
+
+    ReadonlyBytes raw() const { return m_raw; }
+
+    static Address from_flatptr(FlatPtr flatptr)
+    {
+        BigEndian<FlatPtr> big_endian_flatptr { flatptr };
+
+        Address address;
+        address.m_raw.resize(sizeof(FlatPtr));
+        __builtin_memcpy(address.m_raw.data(), &big_endian_flatptr, sizeof(big_endian_flatptr));
+        return address;
+    }
+
+    ErrorOr<FlatPtr> as_flatptr() const
+    {
+        if (m_raw.size() == sizeof(u32))
+            return *bit_cast<BigEndian<u32>*>(m_raw.data());
+        if (m_raw.size() == 2 * sizeof(u32))
+            return *bit_cast<BigEndian<u64>*>(m_raw.data());
+        return ERANGE;
+    }
+
+private:
+    Vector<u8, 4 * sizeof(u32)> m_raw;
```

Comment:
FixedArray should fit a bit better here

Reply:
I use `Vector`s because `FixedArray`s don't have an inline capacity.

Reply:
Then maybe ByteBuffer?
Vector feels wrong, but works I guess

Reply:
Then I would have to use `Detail::ByteBuffer`. `::ByteBuffer` is just an alias for `Detail::ByteBuffer<32>`.
An inline capacity of 32 seems wasteful.

Reply:
I could move `ByteBuffer` out of `Detail` and change it to have a default value (Ali suggested that in the PR that introduced the current `ByteBuffer` as well: https://github.com/SerenityOS/serenity/pull/7151#discussion_r633052207)

I'm not sure if this would cause any problems. But we would have to replace all current occurrences of `ByteBuffer` with `ByteBuffer<>`, which probably would be annoying for cherry-picking ladybird commits I think (@nico)

Reply:
Vector<a, size> seems fine to me fwiw.